### PR TITLE
Abstract classes

### PR DESCRIPTION
**Pull request by dITySoftware**

### CODE REVIEW

File: source/root/usr/lib/logdata-anomaly-miner/aminer/parsing/MultiLocaleDateTimeModelElement.py [hunk 1]
```diff
@@ -296,7 +300,7 @@ def add_format(self, format_string, format_locale, format_timezone):
         if end_separator is not None:
             end_separator = end_separator.encode()
 
-        next_component = self.next_components.get(lookup_key, None)
```
Comment (landauermax) [inline source/root/usr/lib/logdata-anomaly-miner/aminer/parsing/MultiLocaleDateTimeModelElement.py:299]: Why was this deleted? I think it is necessary for the if statement in the next line.

**Review by landauermax — CHANGES_REQUESTED** [review]
Please respond to comment

File: source/root/usr/lib/logdata-anomaly-miner/aminer/parsing/MultiLocaleDateTimeModelElement.py (hunk 1 above)
Comment (dITySoftware) [reply source/root/usr/lib/logdata-anomaly-miner/aminer/parsing/MultiLocaleDateTimeModelElement.py:299]: I don't know why the whole line is highlighted red, but actually only the **None** was deleted. Deepsource has an Issue when the default is set to None in dict.get() as this is the default of the method.

**Review by landauermax — APPROVED** [review]
Looks good to me
